### PR TITLE
[MM-17741] when converting bot to a user set the correct email.

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
When a bot is converted to a user, via the CLI, the email was not getting stored correctly.  For bots, the previously stored email is `<botname>@localhost`.  This solution bypasses setting the user email to the previously stored email, for bots only.

I was able to test this code on my local machine.  When I login as the converted user, with Email Verification Required, I get the notification that the verification email was sent.  However, I'm hoping this is a side effect of running on localhost??

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-17741

### CODE REVIEW

**Comment by jfrerich** [comment]
@DHaussermann, I added you to the ticket because it is due today and this should keep you in the loop faster than the Jira Ticket.  We can setup a spinwick instance, but I am not sure how to ssh to it so you could run the CLI.

**Comment by DHaussermann** [comment]
@jfrerich I'd be okay with testing this one post-merge given how much effort it was to repro. We would need to sort out that config required on the SpinWick for the test to be conclusive right?

File: app/user.go [hunk 1]
```diff
@@ -1105,7 +1105,9 @@ func (a *App) UpdateUser(user *model.User, sendNotifications bool) (*model.User,
 			return nil, model.NewAppError("UpdateUser", "store.sql_user.update.email_taken.app_error", nil, "user_id="+user.Id, http.StatusBadRequest)
 		}
 
-		user.Email = prev.Email
+		if !user.IsBot {
```
Comment (crspeller) [inline app/user.go]: Can we leave a comment here as to why this line exists? This whole function is a bit confusing and this just adds to it.

Comment (jfrerich) [reply app/user.go]: New comment added. Let me know if that seems reasonable. And I agree this was very confusing

**Comment by jfrerich** [comment]
> @jfrerich I'd be okay with testing this one post-merge given how much effort it was to repro. 
Ok.  I just wanted to make sure you were up to date with the latest status.

>We would need to sort out that config required on the SpinWick for the test to be conclusive right?
You would be able to set the config using the UI, but I don't know how you could ssh to run the CLI via a terminal

File: app/user.go (hunk 1 above)
Comment (crspeller) [reply app/user.go]: "To allow email verification..." is that really what we area allowing here? I seems like we are allowing bot account email changes not email verification.

Comment (jfrerich) [reply app/user.go]: That's true.  I'll reword and re-commit.

**Comment by jfrerich** [comment]
@lieut-data This will not reproduce if email verification is not enabled.  Without entering the block, `user.Email` is will not get set to `prev.Email`

**Comment by jfrerich** [comment]
@lieut-data, Yes, I will add some unit tests and get back soon so this can get merged for 5.14!

**Comment by jfrerich** [comment]
@crspeller @lieut-data I have added unit tests for cases when RequireEmailVerificaiton is both true and false.  In both tests for bots, the returned `user.Email` from `UpdateUser` will be the same as the `user.Email` object passed into the method.

**Pull request merged**